### ISSUE TITLE
Multi-threaded stress tests - Assert failure in ebpfcore.sys

### ISSUE THREAD

**Issue by dv-msft** [microsoft/ebpf-for-windows#2258]
### Describe the bug

```eebpfcore.sys``` throws up an ```assert``` failure when subjected to a 'multi-threaded stress test' application using a native ebpf program.

### Steps taken to reproduce bug

Make sure all relevant 'eBPF-For-Windows' (debug) binaries are available on the system under test (Install if components are not present).
Run the multi-threaded stress test application like so:
```ebpf_stress_tests_km -jtt=32 -jtd=300 -jer=true -jerd=250 native_load_attach_detach_unload_random_v4_test```

### Expected behavior

The test command should run to completion without errors and show the 'successful test...' message on the console.

### Actual outcome

An ```assert``` is seen in the ```ebpfcore.sys``` driver in less than a minute while running the stress test application.

### Additional details

Stack trace:

```
*** Assertion failed: !"Thread entry not found"
***   Source File: D:\wrk\ebpf-for-windows\libs\platform\ebpf_epoch.c, line 730

Break repeatedly, break Once, Ignore, terminate Process, or terminate Thread (boipt)? o
Execute '.cxr FFFFF1020DF4E390' to dump context
Break instruction exception - code 80000003 (first chance)
nt!RtlAssert+0xee:
fffff801`435e77ae cc              int     3
2: kd> k
 # Child-SP          RetAddr               Call Site
00 fffff102`0df4e340 fffff801`51a3039a     nt!RtlAssert+0xee [minkernel\ntos\rtl\assert.c @ 132] 
01 fffff102`0df4e8a0 fffff801`51a30a1d     ebpfcore!_ebpf_epoch_get_thread_entry+0xda [D:\wrk\ebpf-for-windows\libs\platform\ebpf_epoch.c @ 730] 
02 fffff102`0df4e8f0 fffff801`51a0cf32     ebpfcore!ebpf_epoch_exit+0x8d [D:\wrk\ebpf-for-windows\libs\platform\ebpf_epoch.c @ 379] 
03 fffff102`0df4e960 fffff801`51a03699     ebpfcore!ebpf_core_invoke_protocol_handler+0x542 [D:\wrk\ebpf-for-windows\libs\execution_context\ebpf_core.c @ 2463] 
04 fffff102`0df4ea10 fffff801`471573cb     ebpfcore!_ebpf_driver_io_device_control+0x8e9 [D:\wrk\ebpf-for-windows\ebpfcore\ebpf_drv.c @ 329] 
05 (Inline Function) --------`--------     Wdf01000!FxIoQueueIoDeviceControl::Invoke+0x44 [minkernel\wdf\framework\shared\inc\private\common\FxIoQueueCallbacks.hpp @ 226] 
06 fffff102`0df4ed90 fffff801`471568fa     Wdf01000!FxIoQueue::DispatchRequestToDriver+0x16b [minkernel\wdf\framework\shared\irphandlers\io\fxioqueue.cpp @ 3325] 
07 fffff102`0df4ee00 fffff801`47155994     Wdf01000!FxIoQueue::DispatchEvents+0x2da [minkernel\wdf\framework\shared\irphandlers\io\fxioqueue.cpp @ 3125] 
08 (Inline Function) --------`--------     Wdf01000!FxIoQueue::QueueRequest+0x97 [minkernel\wdf\framework\shared\irphandlers\io\fxioqueue.cpp @ 2367] 
09 (Inline Function) --------`--------     Wdf01000!FxPkgIo::DispatchStep2+0x4c8 [minkernel\wdf\framework\shared\irphandlers\io\fxpkgio.cpp @ 469] 
0a fffff102`0df4eec0 fffff801`4715542d     Wdf01000!FxPkgIo::DispatchStep1+0x554 [minkernel\wdf\framework\shared\irphandlers\io\fxpkgio.cpp @ 324] 
0b fffff102`0df4ef90 fffff801`47150317     Wdf01000!FxPkgIo::Dispatch+0x5d [minkernel\wdf\framework\shared\irphandlers\io\fxpkgio.cpp @ 119] 
0c (Inline Function) --------`--------     Wdf01000!DispatchWorker+0x57 [minkernel\wdf\framework\shared\core\fxdevice.cpp @ 1587] 
0d (Inline Function) --------`--------     Wdf01000!FxDevice::Dispatch+0x70 [minkernel\wdf\framework\shared\core\fxdevice.cpp @ 1601] 
0e fffff102`0df4eff0 fffff801`43308cbe     Wdf01000!FxDevice::DispatchWithLock+0xa7 [minkernel\wdf\framework\shared\core\fxdevice.cpp @ 1445] 
0f (Inline Function) --------`--------     nt!IopfCallDriver+0x59 [minkernel\ntos\io\iomgr\iomgr.h @ 3759] 
10 fffff102`0df4f050 fffff801`43771a6d     nt!IofCallDriver+0x6e [minkernel\ntos\io\iomgr\iosubs.c @ 3299] 
11 (Inline Function) --------`--------     nt!IoCallDriverWithTracing+0x25 [minkernel\ntos\io\iomgr\iomgr.h @ 1425] 
12 (Inline Function) --------`--------     nt!IopCallDriverReference+0xa6 [minkernel\ntos\io\iomgr\iomgr.h @ 4775] 
13 fffff102`0df4f090 fffff801`4376e8a5     nt!IopSynchronousServiceTail+0x1dd [minkernel\ntos\io\iomgr\internal.c @ 8970] 
14 fffff102`0df4f140 fffff801`4376d736     nt!IopXxxControlFile+0x705 [minkernel\ntos\io\iomgr\internal.c @ 11906] 
15 fffff102`0df4f3c0 fffff801`4347a205     nt!NtDeviceIoControlFile+0x56 [minkernel\ntos\io\iomgr\devctrl.c @ 110] 
16 fffff102`0df4f430 00007ff8`0854f934     nt!KiSystemServiceCopyEnd+0x25 [minkernel\ntos\ke\amd64\trap.asm @ 3644] 
17 0000003c`f6bfe7e8 00007ff8`05cb8e9b     0x00007ff8`0854f934
18 0000003c`f6bfe7f0 000001db`adde0da0     0x00007ff8`05cb8e9b
19 0000003c`f6bfe7f8 00007fff`ed4ed0ec     0x000001db`adde0da0
1a 0000003c`f6bfe800 0000003c`f6bfea18     0x00007fff`ed4ed0ec
1b 0000003c`f6bfe808 00007fff`ed4f7996     0x0000003c`f6bfea18
1c 0000003c`f6bfe810 0000003c`f6bfe840     0x00007fff`ed4f7996
1d 0000003c`f6bfe818 00000000`00122400     0x0000003c`f6bfe840
1e 0000003c`f6bfe820 0000003c`f6bfea58     0x122400
1f 0000003c`f6bfe828 00007fff`00000028     0x0000003c`f6bfea58
20 0000003c`f6bfe830 000001db`addb6d80     0x00007fff`00000028
21 0000003c`f6bfe838 00000000`00000048     0x000001db`addb6d80
22 0000003c`f6bfe840 cccccccc`cccccccc     0x48
23 0000003c`f6bfe848 cccccccc`cccccccc     0xcccccccc`cccccccc
24 0000003c`f6bfe850 00000000`00000028     0xcccccccc`cccccccc
25 0000003c`f6bfe858 00007ff8`07ce3d25     0x28
26 0000003c`f6bfe860 00000000`00122400     0x00007ff8`07ce3d25
27 0000003c`f6bfe868 0000003c`f6bfea28     0x122400
28 0000003c`f6bfe870 0000003c`f6bfea30     0x0000003c`f6bfea28
29 0000003c`f6bfe878 00003508`007ebd81     0x0000003c`f6bfea30
2a 0000003c`f6bfe880 000001db`addb6d80     0x00003508`007ebd81
2b 0000003c`f6bfe888 00007fff`00000048     0x000001db`addb6d80
2c 0000003c`f6bfe890 0000003c`f6bfe964     0x00007fff`00000048
2d 0000003c`f6bfe898 00000000`00000000     0x0000003c`f6bfe964
```

**Comment by Alan-Jowett** [comment]
Can you share a dump + symbols please?

Also, can you tell me what operation ID this is?

**Comment by dv-msft** [comment]
Fixed by PR #2262
